spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+            product3.setDerivedProduct(derivedProduct);
+            product3 = ownerProductApi.createProductByOwner(ownerKey, product3);
+            productIdToProduct.put(product3.getId(), product3);
+
+            ProductDTO productVdc = createVDCProduct(client, ownerKey);
+            productIdToProduct.put(productVdc.getId(), productVdc);
+            ProductDTO productDc = productVdc.getDerivedProduct();
+            productIdToProduct.put(productDc.getId(), productDc);
+
+            // this is for the update process
+            ProductDTO productUp = ownerProductApi.createProductByOwner(ownerKey, Products.random());
+            productIdToProduct.put(productUp.getId(), productUp);
+
+            ContentDTO content1 = Content.random()
+                .metadataExpire(6000L)
+                .requiredTags("TAG1,TAG2");
+            content1 = ownerContentApi.createContent(ownerKey, content1);
+
+            ContentDTO archContent = Content.random()
+                .metadataExpire(6000L)
+                .contentUrl("/path/to/arch/specific/content")
+                .requiredTags("TAG1,TAG2")
+                .arches("i386,x86_64");
+            archContent = ownerContentApi.createContent(ownerKey, archContent);
+
+            ownerProductApi.addContent(ownerKey, product1.getId(), content1.getId(), true);
+            ownerProductApi.addContent(ownerKey, product2.getId(), content1.getId(), true);
+            ownerProductApi.addContent(ownerKey, product2.getId(), archContent.getId(), true);
+            ownerProductApi.addContent(ownerKey, derivedProduct.getId(), content1.getId(), true);
+
+            List<ProductDTO> poolProducts =
+                List.of(product1, product2, virtProduct, product3, productUp, productVdc);
+            Map<String, PoolDTO> poolIdToPool =
+                createPoolsForProducts(ownerApi, ownerKey, poolProducts, brandings);
+
+            consumer.setFacts(Map.of("distributor_version", "sam-1.3"));
+            ReleaseVerDTO releaseVer = new ReleaseVerDTO()
+                .releaseVer("");
+            consumer.setReleaseVer(releaseVer);
+            consumerApi.updateConsumer(consumer.getUuid(), consumer);
+            consumer = consumerApi.getConsumer(consumer.getUuid());
+
+            bindPoolsToConsumer(consumerApi, consumer.getUuid(), poolIdToPool.keySet());
+
+            cdn = cdnApi.createCdn(Cdns.random());
+        }
+    }
+
+    private File createExport(ApiClient apiClient, String consumerUuid, CdnDTO cdn) throws ApiException {
+        String cdnLabel = cdn == null ? null : cdn.getLabel();
+        String cdnName = cdn == null ? null : cdn.getName();
+        String cdnUrl = cdn == null ? null : cdn.getUrl();
+        File export = apiClient.consumers().exportData(consumerUuid, cdnLabel, cdnName, cdnUrl);
+        export.deleteOnExit();
+
+        return export;
+    }
+
+    private File createExportAsync(ApiClient apiClient, String consumerUuid)
```

Comment:
@Ceiu @JoshAlbrecht23 This and the above method (createExport) look generic enough to me to be added to the ExporterUtil class for use by the import tests that are to be imported later

Reply:
Looking at this again, the createExport method is too small to have value porting.
The createExportAsync does not seem to be used anywhere, for now. The ruby tests were all running twice, once using the async and once using the sync endpoints, but I don't think need that for now. Looking at hosted logs and katello code, the async has never been used. Given that, I think we should do what @Ceiu proposed yesterday, which is to leave most tests here to continue using the sync endpoint, but make only one of them use the async one, to make sure it's functional for now. (in the mean time we can start discussions on getting Satellite to using the async one)

Reply:
I added the test 'shouldExportProductsAsync' to verify that the createExportAsync method is working properly!